odxtools/communicationparameter.py

```diff
@@ -1,26 +1,29 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2022 MBition GmbH
 
-from typing import Optional, Any, Union, List, TYPE_CHECKING
 import warnings
+from typing import Any, List, Optional, Union
 
 from odxtools.exceptions import OdxWarning
 
-from .odxlink import OdxLinkDatabase, OdxLinkRef, OdxDocFragment
+from .comparam_subset import (BaseComparam, Comparam, ComplexComparam,
+                              ComplexValue, read_complex_value_from_odx)
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkRef
 from .utils import read_description_from_odx
-from .comparam_subset import BaseComparam, Comparam, ComplexComparam, ComplexValue, read_complex_value_from_odx
 
 
 class CommunicationParameterRef:
 
     def __init__(self,
                  value : Union[str, ComplexValue],
                  id_ref: OdxLinkRef,
+                 is_functional = False,
```

Comment:
please add type annotations wherever possible (everywhere):
```suggestion
                 is_functional: bool = False,
```
also, where is this parameter comming from? (I cannot find it in the XSD for COMPARAM-REF...)

finally -- if this parameter is optional (i.e., not defining it means "I don't care" -- this should probably be
```suggestion
                 is_functional: Optional[bool] = None,
```

Reply:
The parameter is **not** coming from the XSD. The pdx files I have worked with so far group normal comparams in a DiagLayer with `variant_type="PROTOCOL"` whereas comparams for functional addressing are in a separate DiagLayer with `variant_type="FUNCTIONAL_GROUP"`. Without this change there is no way to distinguish between functional addressing non-functional addressing (which is IMO the more frequent use case).

Reply:
ah, okay. I don't really like the fact that after this, objects of the class will contain attributes that are specified by the DiagLayer, but I don't see any other convenient way out either. let's keep this construct, then.